ci-operator/config/openshift/openshift-azure/master.json

```diff
@@ -5,33 +5,48 @@
     "name": "plugin-base"
   },
   "base_images": {
-    "base": {
+    "plugin-base": {
       "cluster": "https://api.ci.openshift.org",
       "namespace": "azure",
       "name": "plugin-base",
       "tag": "latest"
+    },
+    "test-base": {
+      "cluster": "https://api.ci.openshift.org",
+      "namespace": "azure",
+      "name": "test-base",
+      "tag": "latest"
     }
   },
   "test_base_image": {
     "cluster": "https://api.ci.openshift.org",
-    "namespace": "openshift",
-    "name": "release",
-    "tag": "golang-1.10"
+    "namespace": "azure",
+    "name": "test-base",
+    "tag": "latest"
   },
 
   "canonical_go_repository": "github.com/openshift/openshift-azure",
 
-  "binary_build_commands": "make sync",
+  "binary_build_commands": "make build sync",
 
   "images": [
     {
-      "from": "base",
+      "from": "plugin-base",
       "to": "sync",
       "dockerfile_path": "Dockerfile.sync",
       "inputs": {
         "bin": {"paths": [{"source_path": "/go/src/github.com/openshift/openshift-azure/sync", "destination_dir": "."}]},
         "root": {"as": ["sync"]}
       }
+    },
+    {
+      "from": "test-base",
```

Comment:
Needs to be removed. test-base should be built as part of a separate ci-operator job or build

Reply:
Its "fake" build to overlay source code from PR.
Challenge was all build done for test part lands in imagestream `staging`,
all dockerfile build lands in image stream `stable`.

Template execution by default defaults to `stable` image stream (I think its hardcoded in the code). So to run a test I need to have my test code inside `stable` imagestream... 

Inception

Reply:
Still this image needs to be built as part of a different job.

Reply:
OK, this stinks, I see why you need it. @stevekuznetsov we need to discuss this in the CI meeting today.

Reply:
It does. Workaraound is to use our bin image from `pipeline` section. So our template would look like:
```
 - name: test
      image: registry.svc.ci.openshift.org/ci/pipeline:bin
```
Which is a hack too. A little bit less stinky but a hack.